programs/keeper/Keeper.cpp

```diff
@@ -562,7 +567,7 @@ try
     auto main_config_reloader = std::make_unique<ConfigReloader>(
         config_path,
         extra_paths,
-        config().getString("path", KEEPER_DEFAULT_PATH),
+        getKeeperPath(config()),
```

Comment:
this doesn't work for cases when global `path` is defined but no storage path is present (e.g. you defined `log_storage_disk`)

Reply:
Oh, I missed that keeper already support disks (https://github.com/ClickHouse/ClickHouse/pull/50098)

But, I'm not sure that it worth it, this path is used only for `preprocessed_configs`, plus `ConfigReloader` does not support IDisk interface and it's support will introduce a tiny complexity in the code (plus `ConfigReloader` is in `Common`, while `IDisk` in `Disks`). Having some local path for preprocessed config, does not looks bad.

What do you think?

Reply:
No particular reason, just because there was similar code before, and I just moved it into helper.

P.S. also keeper_embedded.xml does not include `path`